### DISCUSSION TITLE
Cannot update state property via $patch with empty object

### DISCUSSION THREAD

**Discussion by jasonrsadler** [vuejs/pinia#2242] [Help and Questions] [answered]
### Reproduction

NA

### Steps to reproduce the bug

Using Vue 2, Update a state item object to an empty object in a Vue page:

```
const myStore = useMyStore()
mystore.$patch({
     selectedItem: {},
})
```

with store:
```
export const useMyStore = defineStore('myStore', {
    state: () => ({
        selectedItem: {},
.
.
```

### Expected behavior

`selectedItem` becomes an empty object

### Actual behavior

`selectedItem` maintains it previous values

### Additional information

NA

**Comment by andrewevans** [comment]
Same behavior with `"@pinia/nuxt": "^0.5.1", "nuxt": "^3.8.0", "vue": "^3.3.6",`.

UPDATE: I switched to using the 2nd method https://pinia.vuejs.org/core-concepts/state.html#Mutating-the-state and this worked. Not sure why it acts differently but it works.

```
    mystore.$patch((state) => {
        state.selectedItem = {}
    })

```

**Reply by posva** [reply, marked as answer]
`$patch()` tries to recursively merge objects, therefore, an empty object doesn't change anything. Using the function syntax or directly changing the property is the correct way
